tests/test_serialization.py

```diff
@@ -0,0 +1,154 @@
+from typing import Optional, Type
+
+import attrs
+import pytest
+
+from web_poet import (
+    HttpResponse,
+    HttpResponseBody,
+    Injectable,
+    RequestUrl,
+    ResponseUrl,
+    WebPage,
+)
+from web_poet.page_inputs.url import _Url
+from web_poet.serialization import (
+    SerializedData,
+    deserialize,
+    read_serialized_data,
+    register_serialization,
+    serialize,
+    write_serialized_data,
+)
+
+
+def _compare_webpages(p1: WebPage, p2: WebPage) -> None:
+    assert p1.response.body == p2.response.body
+    assert p1.response.status == p2.response.status
+    assert p1.response.headers == p2.response.headers
+    assert p1.response._encoding == p2.response._encoding
+    _compare_urls(p1.response.url, p2.response.url)
+
+
+def _compare_urls(u1: _Url, u2: _Url) -> None:
```

Comment:
_Url has no `__eq__` and different classes being not equal even has its own test.